ykrt/src/compile/jitc_yk/aot_ir.rs

```diff
@@ -1,33 +1,28 @@
-//! Yk's AOT IR deserialiser.
+//! The AOT Intermediate Representation (IR).
 //!
-//! This module contains the data structures for the AOT IR and a parser to read it from its
-//! serialised format.
+//! This is the IR created by ykllvm: this module contains both the IR itself and a deserialiser
+//! from ykllvm's output into the IR itself. The IR can feel a little odd at first because we store
+//! indexes into vectors rather than use direct references. This allows us to squeeze the amount of
+//! the memory used down (and also bypasses issues with representing graph structures in Rust, but
```

Comment:
A nitpick that may or may not be worth fixing: squeeze the amount of ~~the~~ memory used down.

Reply:
By the way, I've made no attempt to make the AOT IR small. The true reason for the indices is because you can't serialise references.

Reply:
Since the index types are smaller than a word, we get squeeziness for free :) FWIW, I think the "the" thing is probably below our current threshold for change, but happy to go with what you think.

Reply:
I was about to agree, but I've just remembered that in the AOT IR, the indices are all the same size as a pointer, so there is no space saving.

E.g.
```
pub(crate) struct FuncIdx(usize);
```

I think(?) for the AOT IR you don't really want reduced index sizes: since we encode a whole CFG, you could much more easily blow the limits of (say) 16/24-bit indices.

So I guess in light of that, we should delete the part about space saving?

Reply:
Agreed. Fixed in https://github.com/ykjit/yk/pull/1156/commits/86d540ea8dfb2d3c41f8225a09fa66147a976fc2.